File: general_programming_resources/06_debugging_methodology/examples/stack_trace_bug/ScoutingReport.java
import java.util.HashMap;
import java.util.Map;

public class ScoutingReport {
    public static void main(String[] args) {
        Map<String, String> teamLookup = new HashMap<>();
        teamLookup.put("1515", "Mortorq");
        teamLookup.put("254", "The Cheesy Poofs");

        System.out.println(formatEntry(teamLookup, "1515", "Fast cycle times, reliable auto"));
        System.out.println(formatEntry(teamLookup, 254, "Consistently high scoring"));
    }

    static String formatEntry(Map<String, String> teamLookup, Object teamNumber, String notes) {
        String name = getTeamName(teamLookup, teamNumber);
        return name.toUpperCase() + ": " + notes;
    }

    static String getTeamName(Map<String, String> teamLookup, Object teamNumber) {
        return teamLookup.get(teamNumber);
    }
}

File: general_programming_resources/06_debugging_methodology/examples/stack_trace_bug/scouting_report.cpp
#include <iostream>
#include <map>
#include <string>

std::string* getTeamName(std::map<std::string, std::string>& teamLookup, const std::string& teamNumber) {
    auto it = teamLookup.find(teamNumber);
    if (it == teamLookup.end()) {
        return nullptr;
    }
    return &it->second;
}

std::string formatEntry(std::map<std::string, std::string>& teamLookup, const std::string& teamNumber, const std::string& notes) {
    std::string* name = getTeamName(teamLookup, teamNumber);
    return *name + ": " + notes;
}

int main() {
    std::map<std::string, std::string> teamLookup;
    teamLookup["1515"] = "Mortorq";
    teamLookup["254"] = "The Cheesy Poofs";

    std::cout << formatEntry(teamLookup, "1515", "Fast cycle times, reliable auto") << std::endl;
    std::cout << formatEntry(teamLookup, "0254", "Consistently high scoring") << std::endl;

    return 0;
}
